src/plugins/info/info_impl.cpp

```diff
@@ -167,63 +157,63 @@ void InfoImpl::process_flight_information(const mavlink_message_t& message)
     _flight_information_received = true;
 }
 
-void InfoImpl::translate_binary_to_str(
-    uint8_t* binary, unsigned binary_len, char* str, unsigned str_len)
+std::string InfoImpl::translate_binary_to_str(uint8_t* binary, unsigned binary_len)
 {
+    std::string str(binary_len * 2, '0');
+
     for (unsigned i = 0; i < binary_len; ++i) {
         // One hex number occupies 2 chars.
         // The binary is in little endian, therefore we need to swap the bytes for us to read.
-        snprintf(&str[i * 2], str_len - i * 2, "%02x", binary[binary_len - 1 - i]);
+        snprintf(&str[i * 2], str.length() - i * 2, "%02x", binary[binary_len - 1 - i]);
```

Comment:
Now that I think about it I'm not sure about little endian vs. big endian here. Because for the `uid2` it's basically just an array of bytes while for `uid` it might have had a certain endian type because it was a `uint64_t`.

Reply:
The previous implementation was swapping for git hashes, but not for uid2. I thought it was not consistent, but I messed up probably. I'll revert that.